runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/BatchFlinkExecutableStageContext.java

```diff
@@ -54,7 +61,26 @@ private BatchFlinkExecutableStageContext(JobBundleFactory jobBundleFactory) {
   @Override
   public StateRequestHandler getStateRequestHandler(
       ExecutableStage executableStage, RuntimeContext runtimeContext) {
-    return FlinkBatchStateRequestHandler.forStage(executableStage, runtimeContext);
+    MultimapSideInputHandlerFactory sideInputHandlerFactory =
+        FlinkBatchSideInputHandlerFactory.forStage(executableStage, runtimeContext);
+    ExecutableProcessBundleDescriptor processBundleDescriptor;
+    try {
+      // NOTE: We require an executable bundle descriptor for the StateRequestHandlers construction
+      // below. This only uses the bundle descriptor for side input specs and effectively ignores
+      // data and state endpoints. We rely on the fact that PCollections and coders are structurally
+      // identical between instantiations here to prevent having to wire the original executable
+      // bundle descriptor here. The correct long-term fix is to move side input logic out of
```

Comment:
I don't quite understand this note. What is the problem with the current code, and which side input logic are you referring to that must be moved into ExecutableStage? Would that require also changing ExecutableStagePayload?

Reply:
Basically, the problem is that `ExecutableProcessBundleDescriptor` does not do quite what we want right now. It was intended to be an instantiation `ExecutableStage` with the gRPC endpoints filled out. All the other logic should go into `ExecutableStage` itself. However, we now have side input logic and coder instantiation on EPBD in addition to gRPC endpoints. This general pipeline logic would be better refactored into `ExecutableStage` because we don't have access to the data or state API endpoints here and cannot fill those out.

Luckily, this is not a matter of correctness because EPBD construction (at least for the component ids that we care about) is deterministic. It's just strange to pass in API descriptors where we do not have them and where they will not be used.

Reply:
OK I looked at the code of EPBD, and it seems that extracting that logic would take up less code in the current PR than taken up by this comment ;) So I recommend to do that in this PR.